tests/integration/utils.py

```diff
@@ -353,15 +353,24 @@ def assert_expected_files(source_path, expected_files, tmpdir):
                 package_root_dir = test_path
             else:
                 archive_path = test_path
-                shutil.unpack_archive(archive_path, deps_data_path)
+                if archive_path.endswith(".gem"):
+                    shutil.unpack_archive(archive_path, deps_data_path, "gztar")
+                else:
+                    shutil.unpack_archive(archive_path, deps_data_path)
                 # deps_data_path is unique and contains only one expected package
-                package_root_dir = os.path.join(deps_data_path, os.listdir(deps_data_path)[0])
+                if deps_data_path.endswith(".gem"):
+                    package_root_dir = deps_data_path
+                else:
+                    package_root_dir = os.path.join(deps_data_path, os.listdir(deps_data_path)[0])
```

Comment:
The comment about containing only one package probably belongs to the code that does `listdir()[0]`